Snowflake.Data.Tests/SFConnectionPoolT.cs

```diff
@@ -598,5 +727,35 @@ static async Task InvalidConnectionTaskAsync(string connectionString)
                 await Task.Delay(100);
             }
         }
+        
+        private class TestSnowflakeDbConnection : SnowflakeDbConnection
+        {
+            protected override DbProviderFactory DbProviderFactory => new TestSnowflakeDbFactory();
+        }
+        
+        private class TestSnowflakeDbFactory : DbProviderFactory 
+        {
+            public override SnowflakeDbCommand CreateCommand()
+            {
+                var commandThrowingExceptionOnlyForRollback = new Mock<SnowflakeDbCommand>().As<IDbCommand>();
+                commandThrowingExceptionOnlyForRollback.CallBase = true;
+                commandThrowingExceptionOnlyForRollback.SetupSet(it => it.CommandText = "ROLLBACK")
+                    .Throws(new SnowflakeDbException(SFError.INTERNAL_ERROR, "Unexpected failure on transaction rollback when connection is returned to the pool with pending transaction"));
+                return (SnowflakeDbCommand)commandThrowingExceptionOnlyForRollback.Object;
+            }
+        }
+
+        private readonly struct PoolConfig {
```

Comment:
this struct was already defined in different file in this review - why don't reuse it if necessary?